src/v/cloud_topics/level_zero/stm/types.h

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2025 Redpanda Data, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/redpanda-data/redpanda/blob/master/licenses/rcl.md
+ */
+
+#pragma once
+
+#include "model/fundamental.h"
+
+#include <seastar/core/rwlock.hh>
+
+namespace experimental::cloud_topics {
+
+enum class ctp_stm_key : uint8_t {
+    advance_reconciled_offset = 1,
+};
+
+struct [[nodiscard]] cluster_epoch_fence {
```

Comment:
an object that the write path holds when it wants to replicate batch that will return this structure and the caller will hold this to implement fencing.

term: not used. can remove until we end up using it.